Applications/ResNet/jni/main.cpp

```diff
@@ -0,0 +1,421 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jijoong Moon <jijoong.moon@samsung.com>
+ *
+ * @file	main.cpp
+ * @date	27 Jan 2021
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is Resnet18 Example
+ *
+ */
+
+#include "bitmap_helpers.h"
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <stdlib.h>
+
+#include "databuffer.h"
+#include "databuffer_func.h"
+#include "neuralnet.h"
+#include "nntrainer_error.h"
+#include "tensor.h"
+#include <vector>
+
+/**
+ * @brief     Data size for each category
+ */
+const unsigned int num_class = 10;
+
+const unsigned int num_train = 100;
+
+const unsigned int num_val = 20;
+
+const unsigned int batch_size = 128;
+
+const unsigned int feature_size = 3072;
+
+unsigned int train_count = 0;
+unsigned int val_count = 0;
+
+int width = 32;
+
+int height = 32;
+
+int channel = 3;
+
+unsigned int seed;
+
+std::string resource;
+
+float training_loss = 0.0;
+float validation_loss = 0.0;
+float last_batch_loss = 0.0;
+
+typedef struct {
+  unsigned int remain;
+  std::vector<unsigned int> duplication;
+} counting_info_s;
```

Comment:
Instead of duplication vector shuffle the index vector might also a alternative way to handle the random data I guess :)

Like

vector<int> index;
std::iota(index.begin(), index.end(), 0);
std::shuffle(index.begin(), index.end(), rand_engine);

This way can prevent erase, clear, resize the vector